### PR TITLE
Feat/multi investigation actions

### PR DESCRIPTION
**Pull request by NaamanTsur**
This PR implements updating all investigations under the same group when changing desk or investigator by selecting rows and using the footer actions.

### CODE REVIEW

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 1]
```diff
@@ -335,8 +354,27 @@ const InvestigationTable: React.FC = (): JSX.Element => {
...
+                                const enterDeskTitle = `<p>האם אתה בטוח שאתה רוצה לבחור את דסק <b>${newSelectedDesk?.deskName}</b>?</p>`;
+                                if (newSelectedDesk?.deskName !== indexedRow.investigationDesk) {
+                                    const result = await alertWarning(indexedRow.investigationDesk ? switchDeskTitle : enterDeskTitle, {
+                                        showCancelButton: true,
+                                        cancelButtonText: 'לא',
+                                        cancelButtonColor: theme.palette.error.main,
+                                        confirmButtonColor: theme.palette.primary.main,
+                                        confirmButtonText: 'כן, המשך',
+                                    });
+
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:368]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 2]
```diff
@@ -335,8 +354,27 @@ const InvestigationTable: React.FC = (): JSX.Element => {
...
+                                    if(result.isConfirmed) {
+                                        indexedRow.groupId ?
+                                            await changeGroupsDesk([indexedRow.groupId], newSelectedDesk) :
+                                            await changeInvestigationsDesk([indexedRow.epidemiologyNumber], newSelectedDesk);
+                                        fetchTableData();
+                                    }
+
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:375]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts [hunk 3]
```diff
@@ -29,39 +49,29 @@ const useInvestigationTableFooter = (parameters: InvestigationTableFooterParamet
...
+    const handleConfirmDesksDialog = async (updatedDesk: Desk, transferReason: string) => {
+        const { uniqueGroupIds, epidemiologyNumbers } =
+            checkedIndexedRows.reduce<{
+                uniqueGroupIds: string[],
+                epidemiologyNumbers: number[]
+            }>(toUniqueIdsAndEpidemiologyNumbers, {
+                uniqueGroupIds: [],
+                epidemiologyNumbers: []
+            });
+
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts:61]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts [hunk 4]
```diff
@@ -71,28 +81,29 @@ const useInvestigationTableFooter = (parameters: InvestigationTableFooterParamet
         onDialogClose();
     }
 
-    const handleConfirmInvestigatorsDialog = (updatedIvestigator: InvestigatorOption, transferReason: string) => {
-        const changeInvestigatorLogger = logger.setup({
-            workflow: 'Switch investigator',
-            investigation: checkedRowsIds.join(', '),
-            user: userId
+    const handleConfirmInvestigatorsDialog = async (updatedIvestigator: InvestigatorOption, transferReason: string) => {
+
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts:85]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 5]
```diff
@@ -610,60 +595,47 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
-    const onDeskChange = async (indexedRow: IndexedInvestigation, newSelectedDesk: Desk | null) => {
-        const switchDeskTitle = `<p>האם אתה בטוח שאתה רוצה להחליף את דסק <b>${indexedRow.investigationDesk}</b> בדסק <b>${newSelectedDesk?.deskName}</b>?</p>`;
-        const enterDeskTitle = `<p>האם אתה בטוח שאתה רוצה לבחור את דסק <b>${newSelectedDesk?.deskName}</b>?</p>`;
+    const changeGroupsDesk = async (groupIds: string[], newSelectedDesk: Desk | null, transferReason?: string) => {
+        const changeDeskLogger = logger.setup({
+            workflow: 'Change Groups Desk',
+            user: user.id,
+        });
+        const joinedGroupIds = groupIds.join(', ');
+
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:604]: .

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 6]
```diff
@@ -610,60 +595,47 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
+        changeDeskLogger.info(`performing desk change request for group ${joinedGroupIds}`, Severity.LOW);
+        try {
+            axios.post('/landingPage/changeGroupDesk', {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:607]: don't you need awit before the axios

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

**Pull request merged**